core/aggregation.go

```diff
@@ -32,13 +33,31 @@ type SigningMessage struct {
 	Err                  error
 }
 
-// SignatureAggregation contains the results of aggregating signatures from a set of operators
+// QuorumAttestation contains the results of aggregating signatures from a set of operators by quorums
+type QuorumAttestation struct {
+	// QuorumAggPubKeys contains the aggregated public keys for all of the operators each quorum,
+	// including those that did not sign
+	QuorumAggPubKey map[QuorumID]*G1Point
+	// SignersAggPubKey is the aggregated public key for all of the operators that signed the message for each quorum,
+	// further aggregated across the quorums; operators signing for multiple quorums will be included in
+	// the aggregation multiple times
+	SignersAggPubKey map[QuorumID]*G2Point
+	// AggSignature is the aggregated signature for all of the operators that signed the message for each quorum, mirroring the
+	// SignersAggPubKey.
+	AggSignature map[QuorumID]*Signature
+	// QuorumResults contains the quorum ID and the amount signed for each quorum
+	QuorumResults map[QuorumID]*QuorumResult
+	// SignerMap contains the operator IDs that signed the message
+	SignerMap map[OperatorID]bool
```

Comment:
This is not per quorum, should it be left at in SignatureAggregation?

Reply:
Yeah it's a bit awkward because it's the only field not divided into quorums. But it's easier to track this map at the stage of receiving signatures. I added a comment in the struct to clarify what this field is